src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -1098,23 +1099,50 @@ trait SilverTransforms extends SparkSessionWrapper {
       .withColumn("target_num_workers", last('target_num_workers, true).over(stateUnboundW))
       .withColumn("current_num_workers", last('current_num_workers, true).over(stateUnboundW))
       .withColumn("unixTimeMS_state_start", 'timestamp)
-      .withColumn("unixTimeMS_state_end", coalesce( // if state end open, use pipelineSnapTime, will be merged when state end is received
-        lead('timestamp, 1).over(stateUnboundW) - lit(1), // subtract 1 millis
-        lit(untilTime.asUnixTimeMilli)
-      ))
+      //Change for Issue-846
+
+    val clusterBaseDF = clusterBase(auditLogDF)
+    val clustersRemoved = clusterBaseDF
+      .filter('actionName.isin("permanentDelete"))
+      .select('organization_id, 'cluster_id, 'userEmail.alias("deleted_by"))
+
+    val removedClusterID = clustersRemoved.select("cluster_id").distinct().collect().map(f=>f(0))
+    val w = Window.partitionBy('organization_id, 'cluster_id).orderBy('unixTimeMS_state_start)
+    val timeStampForRemovedCluster = clusterEventsBaselineFinal
+      .drop("driverSpecs", "workerSpecs")
+      .filter(col("cluster_id").isin(removedClusterID: _*))
+      .withColumn("positiveTestCase", when(last('state).over(w) === "RUNNING", lit(true)).otherwise(lit(false)))
+      .withColumn("positiveTestCase2", lead('positiveTestCase,1).over(w))
+      .filter('positiveTestCase2.isNull)
+      .filter('state === "RUNNING").select("timestamp").collect().map(t=>t(0))
```

Comment:
why do all of this? There's a field later that is just `isRunning` if you move this logic to the end of this function, can you not just use the already-derived `isRunning` field to determine deleted clusters that are running? Otherwise you'll have to recreate all the complex logic to catch all the cases.